### DISCUSSION TITLE
Undefined reference to 'ImGui::CreateContext' (vcpkg/cmake)

### DISCUSSION THREAD

**Discussion by Bekobii** [ocornut/imgui#6833] [New Users Build/Link/Run issues ONLY!]
Hey,

I've been installing ImGUI via VCPKG and always get the same error as stated in the title. 

`undefined reference to ImGui::Begin(char const*, bool*, int)`

This is how I import it via CMake.

```
find_package(glfw3 CONFIG REQUIRED)
find_package(imgui CONFIG REQUIRED)
target_link_libraries(TestApp PRIVATE glfw)
target_link_libraries(TestApp PRIVATE imgui::imgui)
```

GLFW does work but not imgui

**Comment by ocornut** [comment]
I can't answer this myself as I am not familiar with vcpkg and cmake but this basically means you are not properly linking with imgui.
So I imagine this `target_link_libraries(TestApp PRIVATE imgui::imgui)` is not doing what you expect.

**Reply by Bekobii** [reply]
Hey,

Well I thought that the maintainer of the repository did create the package for VCPKG since the latest version on VCPKG is the same as here. I'm pretty sure that the Package is faulty since any other VCPKG works fine and the way I link is the official way to work with VCPKG via CMake but thanks for the answer.

**Reply by ocornut** [reply]
I am not creating nor maintaining the vcpkg packages, but I am confident the packages are not faulty as many people are using them. There must be something else.

**Reply by ocornut** [reply]
I recall the vcpkg default to output statically linked library (as I officially don't recommend dynamically linked ones/DLL for dear imgui) so it may me that in that situation the cmake command may be different for that triplet.
